Source/EasyNetQ/RabbitAdvancedBus.cs

```diff
@@ -646,13 +648,23 @@ public virtual void Dispose()
         {
             if (disposed) return;
 
-            foreach(var eventSubscription in eventSubscriptions)
+            foreach (var eventSubscription in eventSubscriptions)
                 eventSubscription.Dispose();
 
-            consumerFactory.Dispose();
-            clientCommandDispatcher.Dispose();
-            confirmationListener.Dispose();
-            connection.Dispose();
+            if (advancedBusEventHandlers.Connected != null)
```

Comment:
Does these null checks make sense?

Reply:
We have the same checks in the constructor, but I'm not so familiar with events, let me check.

Reply:
I've checked it and they are redundant. Thanks